core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java

```diff
@@ -536,11 +536,12 @@ private TaskInfo doUpdateTask(
                     Set<CatalogHandle> catalogHandles = activeCatalogs.stream()
                             .map(CatalogProperties::catalogHandle)
                             .collect(toImmutableSet());
-                    if (sqlTask.setCatalogs(catalogHandles)) {
+                    if (sqlTask.setCatalogs(catalogHandles) || !sqlTask.catalogsLoaded()) {
```

Comment:
With this change, I don't think setCatalogs needs to have a return value.  The main trigger is now `catalogsLoaded`, so this code would be something like this:
```suggestion
sqlTask.setCatalogs(catalogHandles);
if ( !sqlTask.catalogsLoaded()) {
```